daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-deadletter.md

```diff
@@ -0,0 +1,79 @@
+---
+type: docs
+title: "Dead Letter Topics"
+linkTitle: "Dead Letter Topics"
+weight: 4000
+description: "Use dead letter topics to capture undeliverable messages"
+---
+
+## Introduction
+
+Dead letter topics are used to hold messages that cannot be delivered to any receiver. This eases the pressure on receivers by freeing them from dealing with poison messages, while allowing developers to write code that reads from the dead letter topic and either fix the message or abandon it completely.
+
+When a dead letter topic is set, any message that failed to be delivered to an app for a configured topic is put on the dead letter topic.
+
+Dapr enables dead letter topics for all of it's pub/sub components, even if the underlying system does not support this feature natively.
+
+## Configuring a dead letter topic through YAML
+
+The following YAML shows how to configure a dead letter topic named `poisonMessages` for messages consumed from the `orders` topic.
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Subscription
+metadata:
+  name: order
+spec:
+  topic: orders
+  route: /checkout
+  pubsubname: pubsub
+  deadLetterTopic: poisonMessages
+scopes:
+- checkout
+```
+
+## Configuring a dead letter topic programatically
+
+The JSON returned from the `/subscribe` endpoint shows how to configure a dead letter topic named `poisonMessages` for messages consumed from the `orders` topic.
+
+```javascript
+app.get('/dapr/subscribe', (_req, res) => {
+    res.json([
+        {
+            pubsubname: "pubsub",
+            topic: "orders",
+            route: "/checkout",
+            deadLetterTopic: "poisonMessages"
+        }
+    ]);
+});
+```
+
+## Retries and dead letter topics
+
+By default, when a dead letter topic is set, any failing message immediately goes to the dead letter topic.
+To allow Dapr to retry the message before sending it to the dead letter topic, you can use a [retry resiliency policy]({{< ref "policies.md#retries" >}}).
+
+This example shows how to set a constant retry, with 10 maximum delivery attempts every 5 seconds for the `pubsub` pub/sub component.
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Resiliency
+metadata:
+  name: myresiliency
+spec:
+  policies:
+    retries:
+      pubsubRetry:
+        policy: constant
+        duration: 5s
+        maxRetries: 10
+  targets:
+    components:
+      pubsub:
+        inbound:
+          retry: pubsubRetry
+```
+
```

Comment:
## Configuring a subscription for handling the dead letter topics

Remember to now configure a subscription to handling the dead letter topics. For example you can create another declarative subscription to receive these on the same or a different application. The example below shows the `checkout` application subscribing to the `poisonMessages` topic  with another subscription and sending these to be handled by the `/failedmessages` endpoint.

```yaml
apiVersion: dapr.io/v1alpha1
kind: Subscription
metadata:
  name: deadlettertopics
spec:
  topic: poisonMessages
  route: /failedmessages
  pubsubname: pubsub
scopes:
- checkout
```